src/actors/ActorId.ts

```diff
@@ -17,6 +17,8 @@ export default class ActorId {
   private readonly id: string;
 
   constructor(id: string) {
+    if(id==='') throw new Error("ActorId cannot be empty")
```

Comment:
I would say this is not generic enough. In Javascript we should use `encodeURIComponent` to ensure the strings are valid. Maybe it would be good to use something like this:

```javascript
if (!id) {
    throw new Error("ActorId cannot be empty");
}

id = encodeURIComponent(id);
```

Reply:
Will this cause compatibility issues with other language SDKs?